lewis/core/devices.py

```diff
@@ -111,21 +111,25 @@ class DeviceBuilder(object):
     def __init__(self, module):
         self._module = module
 
-        self._device_types = list(get_members(self._module, is_device).values())
-
-        if not self._device_types:
-            # syslog.warning('No devices discovered in module {}.'.format(module.__name__))
-            pass
-        else:
-            pass
-            # syslog.info('Discovered devices in module {}: {}'.format(
-            #    module.__name__, ','.join([x.__name__ for x in self._device_types])))
-
         submodules = get_submodules(self._module)
 
+        # Devices can be in __init__.py, device.py, or any devices/*.py
+        self._device_types = list(get_members(self._module, is_device).values())
+        self._device_types += list(get_members(submodules.get('device'), is_device).values())
```

Comment:
I think either these two cases should be in `_discover_devices`, like for the interfaces and setups. That would actually make all of these discovery methods consistent.

Looking at the `device` case and the way you shuffled around the files for the current devices I'd even raise the question whether we really need that case, it just adds another "magic name". For simple cases I think providing devices in the  `__init__.py` (which is mandatory as a file anyway) is enough, all other cases can go into `devices`. Plus, were not breaking anything by not adding this possibility.